File: server/src/data/subjects.data.js
// All subject names need to have number in the end converted to arabic numerals
// and tildes removed for better matching
// and lowercase for better matching
import IGFO from './IGFO.data.json' with { type: "json" }
import ILOT from './ILOT.data.json' with { type: "json" }
import INBI from './INBI.data.json' with { type: "json" }
import INCE from './INCE.data.json' with { type: "json" }
import INDU from './INDU.data.json' with { type: "json" }
import INEA from './INEA.data.json' with { type: "json" }
import INFO from './INFO.data.json' with { type: "json" }
import INME from './INME.data.json' with { type: "json" }
import INNI from './INNI.data.json' with { type: "json" }
import INRO from './INRO.data.json' with { type: "json" }
import ITOG from './ITOG.data.json' with { type: "json" }
import LILT from './LILT.data.json' with { type: "json" }
import ICIV from './ICIV.data.json' with { type: "json" }
import LQFB from './LQFB.data.json' with { type: "json" }
import LQUI from './LQUI.data.json' with { type: "json" }
import LIMA from './LIMA.data.json' with { type: "json" }
// IMEI is not included because we couldn't find study plans for it
// ICOM is not included because we couldn't find study plans for it
// LINA is not included because we couldn't find study plans for it
// ICIM is not included because we couldn't find study plans for it
// LIFI is not included because we couldn't find study plans for it


export const subjectsData = {
  INNI: INNI,
  INFO: INFO,
  LILT: LILT,
  ILOT: ILOT,
  INRO: INRO,
  INCE: INCE,
  INEA: INEA,
  INBI: INBI,
  INME: INME,
  INDU: INDU,
  ITOG: ITOG,
  IGFO: IGFO,
  ICIV: ICIV,
  LQFB: LQFB,
  LQUI: LQUI,
  LIMA: LIMA
}

File: chatbot-cucei-assistant/subject_data/subjects.data.py
"""Subject data registry."""

from __future__ import annotations

import json
from pathlib import Path

_BASE_DIR = Path(__file__).resolve().parent

def _load_subject(code: str):
    with (_BASE_DIR / f"{code}.data.json").open("r", encoding="utf-8") as file:
        return json.load(file)


IGFO = _load_subject("IGFO")
ILOT = _load_subject("ILOT")
INBI = _load_subject("INBI")
INCE = _load_subject("INCE")
INDU = _load_subject("INDU")
INEA = _load_subject("INEA")
INFO = _load_subject("INFO")
INME = _load_subject("INME")
INNI = _load_subject("INNI")
INRO = _load_subject("INRO")
ITOG = _load_subject("ITOG")
LILT = _load_subject("LILT")
ICIV = _load_subject("ICIV")
LQFB = _load_subject("LQFB")
LQUI = _load_subject("LQUI")
LIMA = _load_subject("LIMA")

# IMEI is not included because we couldn't find study plans for it
# ICOM is not included because we couldn't find study plans for it
# LINA is not included because we couldn't find study plans for it
# ICIM is not included because we couldn't find study plans for it
# LIFI is not included because we couldn't find study plans for it


subjectsData = {
    "INNI": INNI,
    "INFO": INFO,
    "LILT": LILT,
    "ILOT": ILOT,
    "INRO": INRO,
    "INCE": INCE,
    "INEA": INEA,
    "INBI": INBI,
    "INME": INME,
    "INDU": INDU,
    "ITOG": ITOG,
    "IGFO": IGFO,
    "ICIV": ICIV,
    "LQFB": LQFB,
    "LQUI": LQUI,
    "LIMA": LIMA,
}
